parser.y

```diff
@@ -6009,10 +6013,13 @@ SimpleExpr:
 		if tp.Decimal == types.UnspecifiedLength {
 			tp.Decimal = defaultDecimal
 		}
+		explicitCharset := parser.explicitCharset
+		parser.explicitCharset = false
```

Comment:
why change the variable in parser?

Reply:
A simple way to pass the message(explicit set charset)

Reply:
`parser.explicitCharset` is driven by TiDB?

Reply:
No, I add it in this PR